### PR TITLE
:sparkles: ci add retag image workflow

### PR DESCRIPTION
**Pull request by simonhir**
**Description**

Add workflow for retagging image.

### CODE REVIEW

**Review by devtobi — COMMENTED** [review]
Why is this needed? Wouldn't it be better to move this over to https://github.com/it-at-m/.github by providing a reusable workflow and workflow template? I would even wrap all the executed actions inside a custom action.

**Comment by simonhir** [comment]
> Why is this needed?

In example for moving the latest tag if a new image has a bug. Also this would allow to build images with RC suffix and retagging it after tests are successful.

> I would even wrap all the executed actions inside a custom action.

Done

> Wouldn't it be better to move this over to https://github.com/it-at-m/.github

Would test it here (could only be done after merge) and then we can migrate it over

File: .github/workflows/retag_image.yaml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+        description: 'New tag'
+        type: string
+        default: "latest"
+
+jobs:
+  tag_image:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
```
Comment (devtobi) [inline .github/workflows/retag_image.yaml:24]: For consistency I'd even add the checkout action to the custom action itself, as that is what we did in the other custom actions in .github-Repo as well. But I am not 100% sure if this is best practice.

Comment (simonhir) [reply .github/workflows/retag_image.yaml:24]: I left it there as this isn't required for the workflow but only for accessing the action and can be removed after we moved the action to .github.

**Review by devtobi — APPROVED** [review]
LGTM

**Pull request merged**